src/libraries/Native/Unix/System.Native/pal_io.c

```diff
@@ -1263,9 +1263,30 @@ char* SystemNative_RealPath(const char* path)
     return realpath(path, NULL);
 }
 
+static int16_t ConvertLockType(int16_t managedLockType)
+{
+    // the managed enum Interop.Sys.LockType has no 1:1 mapping with corresponding Unix values
+    // which can be different per distro:
+    // https://github.com/torvalds/linux/blob/fcadab740480e0e0e9fa9bd272acd409884d431a/arch/alpha/include/uapi/asm/fcntl.h#L48-L50
+    // https://github.com/freebsd/freebsd-src/blob/fb8c2f743ab695f6004650b58bf96972e2535b20/sys/sys/fcntl.h#L277-L279
+    switch (managedLockType)
+    {
+        case 0:
+            return F_RDLCK;
+        case 1:
+            return F_WRLCK;
+        case 2:
+            return F_UNLCK;
+        default:
+            assert_msg(false, "Unknown Lock Type", (int)managedLockType);
+            return -1;
```

Comment:
What's to guarantee one of these values isn't actually -1?

Given that this is an internal implementation detail (we own all call sites calling this and only pass in known const values), I suggest we just make this:
```C++
case 0:
    return F_RDLCK;
case 1:
    return F_WRLCK;
default:
    assert_msg(managedLockType == 2, "...");
    return F_UNLCK;
```
or something like that.  And then undo the unixLockType < 0 change at the call site to this helper.